### PR TITLE
element metadata access cleanup

### PR DESCRIPTION
**Pull request by pq**
/cc @bwilkerson

### CODE REVIEW

File: lib/src/rules/always_specify_types.dart [hunk 1]
```diff
@@ -59,22 +59,8 @@ main() {
...
-bool _isOptionallyParameterized(TypeParameterizedElement element) {
-  var metadata = element.metadata;
-  return metadata.any((ElementAnnotation a) => _isOptionalTypeArgs(a.element));
-}
-
-bool _isOptionalTypeArgs(Element? element) =>
-    element is PropertyAccessorElement &&
-    element.name == _optionalTypeArgsVarName &&
-    element.library.name == _metaLibName;
+bool _isOptionallyParameterized(TypeParameterizedElement element) =>
```
Comment (bwilkerson) [inline lib/src/rules/always_specify_types.dart:62]: Consider inlining this.

File: lib/src/ast.dart [hunk 2]
```diff
@@ -81,26 +81,12 @@ bool hasConstantError(LinterContext context, Expression node) {
...
-bool hasLiteralAnnotation(Element element) {
-  var metadata = element.metadata;
-  for (var i = 0; i < metadata.length; i++) {
-    if (metadata[i].isLiteral) {
-      return true;
-    }
-  }
-  return false;
-}
+@Deprecated('prefer: element.hasLiteral')
```
Comment (bwilkerson) [inline lib/src/ast.dart:84]: It looks like you already removed references from this package. Is there any reason not to remove these two functions?

**Review by bwilkerson — APPROVED** [review]

File: lib/src/rules/always_specify_types.dart (hunk 1 above)
Comment (pq) [reply lib/src/rules/always_specify_types.dart:62]: Good idea.

File: lib/src/ast.dart (hunk 2 above)
Comment (pq) [reply lib/src/ast.dart:84]: It *might* be used internally?

**Pull request merged**